tasks/main.yml

```diff
@@ -0,0 +1,89 @@
+---
+- name: Create consul group
+  group:
+    name: consul
+    system: yes
+
+- name: Create consul user
+  user:
+    name: consul
+    group: consul
+    createhome: no
+    system: yes
+
+- name: Mount "{{ consul_data_dir }}"
+  mount:
+    name: "{{ consul_data_dir }}"
+    src: "{{ consul_openstack_data_device }}"
+    fstype: ext4
+    state: mounted
+  when: consul_server
+
+- name: Create consul directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: consul
+    group: consul
+    mode: 0750
+  with_items:
+    - "{{ consul_config_dir }}"
+    - "{{ consul_data_dir }}"
+
+- name: Download and unarchive the Consul binary
+  unarchive:
+    src: "{{ consul_url }}"
+    dest: "{{ consul_bin_dir }}"
+    owner: consul
+    group: consul
+    remote_src: yes
+    creates: "{{ consul_bin_dir }}/consul"
+
+- name: Create Consul configuration file
+  template:
+    src: config.json.j2
+    dest: "{{ consul_config_dir }}/config.json"
+    owner: consul
+    group: consul
+
+- name: Create Consul service definition file
+  copy:
+    content: "{{ consul_service_config | to_nice_json(indent=4) }}"
+    dest: "{{ consul_config_dir }}/service.json"
+    owner: consul
+    group: consul
+  when: consul_service_config != ""
+
+- name: Create Consul systemd service file
+  template:
+    src: consul.service.j2
+    dest: /etc/systemd/system/consul.service
+
+- name: Enable and start systemd service to start on boot automatically
+  systemd:
+    name: consul.service
+    enabled: yes
+    state: started
+    daemon_reload: yes
+  notify: restart consul
+
+- name: Open TCP/UDP ports on firewall needed for Consul RPC/gossip
+  ufw:
+    rule: allow
+    port: '8300:8302'
```

Comment:
Technically, non-server nodes only need port 8301, but I don't care strongly whether to change this here – there won't be anything else listening on ports 8300 and 8302, so opening these ports on non-server nodes doesn't do any harm.

Reply:
:+1: